File: mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int no_block;
    do
    {
        no_block = get_int("Height: "); //get height of blocks from user until he inputs 1-8
    }
    while (no_block <= 0 || no_block >= 9);


    int block = 0;
    int space = 0;
    int n = 1;

    while (block < no_block) //loop until blocks is same as height
    {
        space = (no_block - n);
        block = (no_block - space);
        while (0 > space) // loop until there are no more spaces
        {
            printf(" ");
            space--;
        }
        while (0 > block) //loop until there are no more blocks
        {
            printf("#");
            block--;
        }
        printf("\n"); //new line after each iteration
        n++; //increase count
    }
}

File: sentimental-mario-more/mario.py
# create mario more with python

# get value of 1 - 8; reject any other input

while True:
    try:
        no_blocks = int(input("Height: "))
        if no_blocks in range(1, 9):
            break
    except ValueError:
        print("The input was not a valid integer.")

# set variables
block = 0
space = 0
n = 1

while True:
    # get number of blocks and spaces to print
    space = no_blocks - n
    block = no_blocks - space
    left_block, right_block = block, block

    # print space
    while space > 0:
        print(" ", end="")
        space -= 1

    # print blocks
    while left_block > 0:
        print("#", end="")
        left_block -= 1

    print("  ", end="")

    # print blocks
    while right_block > 0:
        print("#", end="")
        right_block -= 1

    # print new line
    print("")
    n += 1

    # loop until all blocks are printed
    if right_block < n:
        break
